rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -385,6 +385,11 @@ def run_all_post_decoding_rules(
             address_counterparty = self.rules.addresses_to_counterparties.get(transaction.to_address)  # noqa: E501
             if address_counterparty is not None:
                 counterparties.add(address_counterparty)
+            else:
+                # If there is no address specific counterparty, check if there are any rules for
+                # all counterparties. This way we can apply post decoding rules to a transaction
+                # irrespective of the to_address.
+                counterparties = {cpt_details.identifier for cpt_details in self.rules.all_counterparties}  # noqa: E501
```

Comment:
this sound wrong because the address_to_counterparty was there to avoid running all the post decoding rules in all the transactions and if I understoo correctly if we get a transaction that is not represented in the rules here you are executing all the rules again (really slow).

In which case do you need it? Isn't there another thing you can check againt? like the tx_type or something

Reply:
I modified it so that it runs only for transactions with type 100 (A deposit of ETH from L1 to L2 via the Arbitrum bridge).